clap_complete/src/dynamic/completer.rs

```diff
@@ -177,19 +177,82 @@ fn complete_arg(
                     if let Some(value) = value {
                         if let Some(arg) = cmd.get_arguments().find(|a| a.get_long() == Some(flag))
                         {
-                            completions.extend(
-                                complete_arg_value(value.to_str().ok_or(value), arg, current_dir)
-                                    .into_iter()
-                                    .map(|comp| {
-                                        CompletionCandidate::new(format!(
-                                            "--{}={}",
-                                            flag,
-                                            comp.get_content().to_string_lossy()
-                                        ))
-                                        .help(comp.get_help().cloned())
-                                        .visible(comp.is_visible())
-                                    }),
-                            );
+                            match arg.get_value_delimiter() {
+                                Some(delim) => {
+                                    let values = {
+                                        let mut delim_buffer = [0; 4];
+                                        let delim = delim.encode_utf8(&mut delim_buffer);
+
+                                        value.split(delim).into_iter().collect::<Vec<_>>()
+                                    };
+                                    let last = values.last();
+                                    let prefix = values
+                                        .split_at(values.len() - 1)
+                                        .0
+                                        .join(OsString::from(delim.to_string()).as_os_str());
+                                    match last {
```

Comment:
Can't we do something like
```rust
let (prefix, value) = arg.get_value_delimiter().and_then(|d| value.split_once(d)).unwrap_or((OsStr::new(), value));
...
```
this abstracts away the knowledge of delimiters.  We just need to preprend `prefix` to each value.